node/podcontroller.go

```diff
@@ -113,6 +113,11 @@ type PodController struct {
 	// From the time of creation, to termination the knownPods map will contain the pods key
 	// (derived from Kubernetes' cache library) -> a *knownPod struct.
 	knownPods sync.Map
+
+	// podForDeletion is also a map of key -> deletablePod. It is used only by the syncHandler to supress
+	// duplicate deletions. It is deleted when the pod is non-gracefully deleted from Kubernetes, in the event
+	// handler.
+	podForDeletion sync.Map
```

Comment:
Again, I'm confused. The deletion event comes from the API when the pod is marked for deletion.

Reply:
Perhaps this is a matter of phrasing. I've attempted to rephrase.